integration-guide.md

```diff
@@ -0,0 +1,161 @@
+
+    passport.serializeUser(function(user, done) {
+      done(null, user);   
+    });
+    passport.deserializeUser(function(user, done) {
+        done(null, user);
+    });
+
+Finally, we specify to passport to use the defined strategy as a middleware. Naming the strategy is optional but good practice as multiple strategies can be defined.
+    passport.use('oidc', strategy);
+
+------
+
+## **`middleware/issuanceSession.js`**
+
+An issuance session identifier is required for user registration using Lissi Connect. The `issuanceSession` act as middleware and connects to the Lissi Connect API to get the required field `issuance-session-id`. The Lissi Connect `POST` defines a body value example as follow
+
+    {
+      "subjectIdentifier": "string",
+      "claims": {
+        "additionalProp1": "string",
+        "additionalProp2": "string",
+        "additionalProp3": "string"
+      },
+      "credentialDefinitionId": "string"
+    }
+
+The subject identifier is a UUID identifier randomly generated using the uuidv4 node package
+
+    const { uuid } = require('uuidv4');
+    const userId = uuid()
+
+> This module will be deprecated in the future in favour of module UUID. Most of the functionality of this module is already included in UUID since version 8.3.0, so most of the functions of this module have already been marked as deprecated.
+
+The claims must follow the Credential definition template related to credentialDefinitionId.
+
+    const issuanceSession = {
+      subjectIdentifier : userId,
+      claims:{
+        "First Name": req.query.firstName,
+        "Last Name": req.query.lastName,
+        "E-Mail": req.query.email,
+        "User ID": userId
+      },
+      credentialDefinitionId: process.env.CREDENTIAL_DEFINITION_ID
+    }
+
+The user's information are retrieved from the request parameter either passed through the URL or the body object depending on the implementation choice.
+
+Afterwards, we construct an Axios `POST` request with the issuanceSession object passed to the body of the request
+
+    const axios = require('axios');
+    const result = await axios.post('https://auth.lissi.io/api/issuance-sessions', issuanceSession)
+    .catch(e =>{
+        console.error(e)
+    })
+
+Before calling on the next middleware function we modify the request object by adding an `issuance-session-id` field to `res.locals` as conventionally advised
+
+    res.locals.isssuance_session_id = results.data.id;
+    next();
+
+------
+
+## **`routes/auth.js`**
+
+The `auth.js` route file relies on both previously detailed files. It makes use of the defined `'oidc'` passport strategy for authenticating the user (Login & Register). Although the register methods rely on the `getIssuanceSession` middleware function to get the required `issuance _session_id` to register a new user through Lissi Connect.
```

Comment:
should it be  `issuance-session-id` instead of `issuance _session_id`?
Or why are you using `-` above and `_` here?

Reply:
In the code the custom parameter required is indeed `issuance _session_id` I'll modify the first occurence

Reply:
Still open right?

Reply:
Fixed and stick with `_`